pallets/payment/src/lib.rs

```diff
@@ -379,25 +402,84 @@ pub mod pallet {
+		#[transactional]
+		#[pallet::weight(T::WeightInfo::request_payment())]
+		pub fn request_payment(
+			origin: OriginFor<T>,
+			from: T::AccountId,
+			asset: AssetIdOf<T>,
+			amount: BalanceOf<T>,
+		) -> DispatchResultWithPostInfo {
+			let to = ensure_signed(origin)?;
+
+			// create PaymentDetail and add to storage
+			<Self as PaymentHandler<T>>::create_payment(
+				from.clone(),
+				to.clone(),
+				asset,
+				amount,
+				None,
+				PaymentState::PaymentRequested,
+				false,
+			)?;
+
+			Self::deposit_event(Event::PaymentRequestCreated { from, to });
+
+			Ok(().into())
+		}
+
+		// This extrinsic allows the sender to fulfill a payment request created by a recipient.
+		// The amount will be transferred to the recipient and payment removed from storage
+		#[transactional]
+		#[pallet::weight(T::WeightInfo::accept_and_pay())]
+		pub fn accept_and_pay(
+			origin: OriginFor<T>,
+			to: T::AccountId,
+		) -> DispatchResultWithPostInfo {
+			let from = ensure_signed(origin)?;
+
+			let payment =
+				Payment::<T>::get(from.clone(), to.clone()).ok_or(Error::<T>::InvalidPayment)?;
+
+			ensure!(payment.state == PaymentState::PaymentRequested, Error::<T>::InvalidAction);
+
+			// reserve all the fees from the sender
+			<Self as PaymentHandler<T>>::reserve_payment_amount(&from, &to, payment)?;
+
+			// release the payment and delete the payment from storage
+			<Self as PaymentHandler<T>>::settle_payment(
+				from.clone(),
+				to.clone(),
+				Percent::from_percent(100),
+			)?;
+
+			Self::deposit_event(Event::PaymentRequestCompleted { from, to });
+
+			Ok(().into())
+		}
 	}
 
 	impl<T: Config> PaymentHandler<T> for Pallet<T> {
-		/// The function will create a new payment. When a new payment is created, an amount + incentive
-		/// is reserved from the payment creator. The incentive amount is reserved in the creators account.
-		/// The amount is transferred to the payment recipent but kept in reserved state. Only when the release action
-		/// is triggered the amount is released to the recipent and incentive released to creator.
+		/// The function will create a new payment. The fee and incentive amounts will be calculated and the
+		/// `PaymentDetail` will be added to storage.
 		#[require_transactional]
 		fn create_payment(
 			from: T::AccountId,
 			recipient: T::AccountId,
 			asset: AssetIdOf<T>,
 			amount: BalanceOf<T>,
 			remark: Option<BoundedDataOf<T>>,
-		) -> DispatchResult {
+			payment_state: PaymentState<T::BlockNumber>,
+			charge_incentive_amount: bool,
+		) -> Result<PaymentDetail<T>, sp_runtime::DispatchError> {
 			Payment::<T>::try_mutate(
 				from.clone(),
 				recipient.clone(),
-				|maybe_payment| -> DispatchResult {
+				|maybe_payment| -> Result<PaymentDetail<T>, sp_runtime::DispatchError> {
 					// ensure a payment is not already in process
 					if maybe_payment.is_some() {
 						// do not overwrite an in-process payment!
```

Comment:
Comments like docs are easy to get outdated and out of sync, I believe there's no released/cancelled states anymore? 😜